Source/ObjectiveDropboxOfficial/Platform/ObjectiveDropboxOfficial_macOS/DBClientsManager+DesktopAuth-macOS.m

```diff
@@ -21,9 +23,28 @@ + (void)authorizeFromControllerDesktop:(NSWorkspace *)sharedApplication
       [[DBDesktopSharedApplication alloc] initWithSharedApplication:sharedApplication
                                                          controller:controller
                                                             openURL:openURL];
+  [DBDesktopSharedApplication setDesktopSharedApplication:sharedDesktopApplication];
   [[DBOAuthManager sharedOAuthManager] authorizeFromSharedApplication:sharedDesktopApplication];
 }
 
++ (void)authorizeFromControllerDesktopV2:(NSWorkspace *)sharedApplication
```

Comment:
Why not channel the `authorizeFromControllerDesktop:controller:openURL` through this new method? seems like duplicate logic.

Reply:
yep, I moved the shared logic into a private method.